### PR TITLE
Add integration tests for Secret Management API

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
**Purpose**
$subject
Related issue - [#12264](https://github.com/wso2/product-is/issues/12264)

### CODE REVIEW

**Comment by ChanikaRuchini** [comment]
Manual test build performed and verified.

**Review by darshanasbg — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementFailureTest.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+        }
+    }
+
+    @Test
+    public void testGetSecretTypeByInvalidName() {
+
+        Response response = getResponseOfGet(SECRET_TYPE_API_BASE_PATH + PATH_SEPARATOR + "randomName");
+
+        validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "SECRETM_00021");
+
```
Comment (sumedhe) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementFailureTest.java:120]: Unnecessary newline

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementFailureTest.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .header(HttpHeaders.LOCATION, notNullValue());
+        String location = response.getHeader(HttpHeaders.LOCATION);
+        assertNotNull(location);
+        secretName = location.substring(location.lastIndexOf("/") + 1);
+        assertNotNull(secretName);
+        response = getResponseOfPost(SECRET_API_BASE_PATH + PATH_SEPARATOR + SECRET_TYPE, body);
+        validateErrorResponse(response, HttpStatus.SC_CONFLICT, "SECRETM_00006");
+
```
Comment (sumedhe) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementFailureTest.java:164]: Unnecessary newline

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementTestBase.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+public class SecretManagementTestBase extends RESTAPIServerTestBase {
+
+    public static final String API_DEFINITION_NAME = "secretManager.yaml";
+    public static final String API_VERSION = "v1";
+    public static final String API_PACKAGE_NAME = "org.wso2.carbon.identity.api.server.secret.management.v1";
+    public static final String SECRET_TYPE_API_BASE_PATH = "/secret-type";
+    public static final String SECRET_API_BASE_PATH = "/secrets";
+    public static final String PATH_SEPARATOR = "/";
+    public static final String SECRET_TYPE = "ADAPTIVE_AUTH_CALL_CHOREO";
+
```
Comment (sumedhe) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementTestBase.java:43]: Unnecessary newline

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementFailureTest.java (hunk 2 above)
Comment (ChanikaRuchini) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementFailureTest.java:164]: Addressed with commit [252de16](https://github.com/wso2/product-is/pull/12546/commits/252de169987046a367e9c09f983731935058c944)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementFailureTest.java (hunk 1 above)
Comment (ChanikaRuchini) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementFailureTest.java:120]: Addressed with commit [252de16](https://github.com/wso2/product-is/pull/12546/commits/252de169987046a367e9c09f983731935058c944)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementTestBase.java (hunk 3 above)
Comment (ChanikaRuchini) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/secret/management/v1/SecretManagementTestBase.java:43]: Addressed with commit [252de16](https://github.com/wso2/product-is/pull/12546/commits/252de169987046a367e9c09f983731935058c944)

**Pull request merged**